src/createServer.js

```diff
@@ -1,8 +1,143 @@
+/* eslint-disable no-console */
+/* eslint-disable curly */
 'use strict';
 
+const http = require('node:http');
+const fs = require('node:fs');
+const path = require('node:path');
+const { formidable } = require('formidable');
+const {
+  createGzip,
+  createBrotliCompress,
+  createDeflate,
+} = require('node:zlib');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    if (req.url === '/favicon.ico') {
+      res.statusCode = 200;
+      res.end('');
+
+      return;
+    }
+
+    if (req.url === '/compress' && req.method === 'GET') {
+      res.statusCode = 400;
+      res.end('');
+
+      return;
+    }
+
+    if (req.url === '/' && req.method === 'GET') {
+      const normalizePath = path.resolve(__dirname, 'index.html');
+
+      res.statusCode = 200;
+      res.setHeader('Content-Type', 'text/html');
+
+      const fileStream = fs.createReadStream(normalizePath);
+
+      fileStream.pipe(res).on('error', () => {});
+
+      fileStream.on('error', (err) => {
+        res.statusCode = 500;
+        res.end(`Server error ${err}`);
+      });
+
+      res.on('close', () => fileStream.destroy());
+
+      return;
+    }
+
+    if (req.method === 'POST' && req.url === '/compress') {
+      const form = formidable({});
+      let fields, files;
+
+      try {
+        [fields, files] = await form.parse(req);
+      } catch (err) {
+        res.writeHead(err.httpCode || 400, { 'Content-Type': 'text/plain' });
+        res.end(String(err));
+
+        return;
+      }
+
+      const compressionType =
+        Array.isArray(fields.compressionType) && fields.compressionType[0];
+
+      const filePath = Array.isArray(files.file) && files.file[0].filepath;
```

Comment:
Ensure that 'files.file' is correctly parsed and validated. If 'files.file' is not an array, this logic might fail. Consider adding additional validation to handle different data structures.